src/Model/User.php

```diff
@@ -304,100 +280,83 @@ public static function create(array $data)
-			VALUES ('%s', '%s', '%s', '%s', '%s', '%s', '%s', '%s', '%s', '%s', '%s', %d, %d, 'UTC', '')",
-			dbesc(generate_user_guid()),
-			dbesc($username),
-			dbesc($new_password_encoded),
-			dbesc($email),
-			dbesc($openid_url),
-			dbesc($nickname),
-			dbesc($pubkey),
-			dbesc($prvkey),
-			dbesc($spubkey),
-			dbesc($sprvkey),
-			dbesc(datetime_convert()),
-			intval($verified),
-			intval($blocked)
-		);
-
-		if ($r) {
-			$r = q("SELECT * FROM `user`
-				WHERE `username` = '%s' AND `password` = '%s' LIMIT 1",
-				dbesc($username),
-				dbesc($new_password_encoded)
-			);
-			if (DBM::is_result($r)) {
-				$u = $r[0];
-				$newuid = intval($r[0]['uid']);
-			}
+		$insert_result = dba::insert('user', [
+			'guid'     => generate_user_guid(),
+			'username' => $username,
+			'password' => $new_password_encoded,
+			'email'    => $email,
+			'openid'   => $openid_url,
+			'nickname' => $nickname,
+			'pubkey'   => $pubkey,
+			'prvkey'   => $prvkey,
+			'spubkey'  => $spubkey,
+			'sprvkey'  => $sprvkey,
+			'verified' => $verified,
+			'blocked'  => $blocked,
+			'timezone' => 'UTC',
+			'register_date' => datetime_convert(),
+			'default-location' => ''
+		]);
+
+		if ($insert_result) {
+			$uid = dba::lastInsertId();
+			$user = dba::select('user', [], ['uid' => $uid], ['limit' => 1]);
 		} else {
-			$result['message'] .= t('An error occurred during registration. Please try again.') . EOL;
-			return $result;
+			throw new Exception(t('An error occurred during registration. Please try again.'));
 		}
 
-		/**
-		 * if somebody clicked submit twice very quickly, they could end up with two accounts
-		 * due to race condition. Remove this one.
-		 */
-		$r = q("SELECT `uid` FROM `user`
-			WHERE `nickname` = '%s' ",
-			dbesc($nickname)
-		);
-		if (DBM::is_result($r) && count($r) > 1 && $newuid) {
-			$result['message'] .= t('Nickname is already registered. Please choose another.') . EOL;
-			dba::delete('user', array('uid' => $newuid));
-			return $result;
+		if (!$uid) {
+			throw new Exception(t('An error occurred during registration. Please try again.'));
 		}
 
-		if (x($newuid) !== false) {
-			$r = q("INSERT INTO `profile` ( `uid`, `profile-name`, `is-default`, `name`, `photo`, `thumb`, `publish`, `net-publish` )
-				VALUES ( %d, '%s', %d, '%s', '%s', '%s', %d, %d ) ",
-				intval($newuid),
-				t('default'),
-				1,
-				dbesc($username),
-				dbesc(System::baseUrl() . "/photo/profile/{$newuid}.jpg"),
-				dbesc(System::baseUrl() . "/photo/avatar/{$newuid}.jpg"),
-				intval($publish),
-				intval($netpublish)
-			);
-			if ($r === false) {
-				$result['message'] .= t('An error occurred creating your default profile. Please try again.') . EOL;
-				// Start fresh next time.
-				dba::delete('user', array('uid' => $newuid));
-				return $result;
-			}
+		// if somebody clicked submit twice very quickly, they could end up with two accounts
+		// due to race condition. Remove this one.
```

Comment:
What about just creating an unique index on the nickname in the user table? This would be easier.

Reply:
Absolutely, it's in my DB suggested changes document.

Reply:
if you are not sure how to apply the changes to dbstructure.php, just ask.

Reply:
I probably won't add anything myself until I submit my document to you.

Reply:
Okay.